shell/browser/api/electron_api_push_notifications_mac.mm

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 Asana, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/api/electron_api_push_notifications.h"
+
+#include <string>
+
+#import "shell/browser/mac/electron_application.h"
+#include "shell/common/gin_converters/value_converter.h"
+
+namespace electron {
+
+namespace api {
+
+void PushNotifications::RegisterForAPNSNotifications() {
+  [[AtomApplication sharedApplication]
+      registerForRemoteNotificationTypes:NSRemoteNotificationTypeBadge |
```

Comment:
This method appears to be [deprecated](https://developer.apple.com/documentation/uikit/uiapplication/1623010-registerforremotenotificationtyp), can we use the [newer version](https://developer.apple.com/documentation/uikit/uiapplication/1623078-registerforremotenotifications?language=objc) instead?

Reply:
When I attempt to use `registerForRemoteNotifications` instead, I see:
```
../../electron/shell/browser/api/electron_api_push_notifications_mac.mm:28:40: error: 'registerForRemoteNotifications' is only available on macOS 10.14 or newer [-Werror,-Wunguarded-availability-new]
  [[AtomApplication sharedApplication] registerForRemoteNotifications];
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
sdk/xcode_links/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSApplication.h:501:1: note: 'registerForRemoteNotifications' has been marked as being introduced in macOS 10.14 here, but the deployment target is macOS 10.13.0
- (void)registerForRemoteNotifications API_AVAILABLE(macos(10.14));
^
../../electron/shell/browser/api/electron_api_push_notifications_mac.mm:28:40: note: enclose 'registerForRemoteNotifications' in an @available check to silence this warning
  [[AtomApplication sharedApplication] registerForRemoteNotifications];
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```